Author	Matt Johnston
Date 	2010-07-21T13:33:07 

 weakness in embedded version of mp_prime_next_prime() opens a security risk

 http://bugs.gentoo.org/show_bug.cgi?id=328383
 http://bugs.gentoo.org/show_bug.cgi?id=328409
 https://bugzilla.redhat.com/show_bug.cgi?id=615088
 Exact effects of the bug are uncertain.

============================================================
--- a/libtommath/bn_mp_prime_next_prime.c	d0f2cc0de0f3f223bdf94c4b717a14875d131614
+++ b/libtommath/bn_mp_prime_next_prime.c	3f93b19342d6ac962e0cedfb8813e1ecbc55c759
@@ -143,7 +143,7 @@ int mp_prime_next_prime(mp_int *a, int t

       /* is this prime? */
       for (x = 0; x < t; x++) {
-          mp_set(&b, ltm_prime_tab[t]);
+          mp_set(&b, ltm_prime_tab[x]);
           if ((err = mp_prime_miller_rabin(a, &b, &res)) != MP_OKAY) {
              goto LBL_ERR;
           }
